test/IntegrationTestCommon/Factories/WebApplicationFactoryBase.cs

```diff
@@ -119,13 +118,48 @@ public void UpdateConfiguration(string key, string? value)
     /// </summary>
     protected override void ConfigureWebHost(IWebHostBuilder builder)
     {
-        if (SqliteConnection == null)
+        if (TestDatabase == null)
         {
-            SqliteConnection = new SqliteConnection("DataSource=:memory:");
-            SqliteConnection.Open();
-            _handleSqliteDisposal = true;
+            TestDatabase = new SqliteTestDatabase();
+            HandleDbDisposal = true;
         }
```

Comment:
Could these just be default values on the class property?

Also, what's the purpose of the `HandleDbDisposal` flag? Why isn't it set if a `TestDatabase` is supplied? (wouldn't it still need to be set up/torn down?)